rbc/irtools.py

```diff
@@ -354,8 +355,7 @@ def compile_to_LLVM(functions_and_signatures,
 
     """
     target_desc = registry.cpu_target
-
-    if target_info is None:
+    if target_info is None and 0:
```

Comment:
If the condition above, you were always using the omniscidb target/typing contexts. I don't think the RemoteJIT should use those contexts since it is a bit less restrict compared to omniscidb one (i.e. it can do anything numba can). Also, if the target info is never `None`, one should add a flag indicating if we're compiling for a RemoteJIT or to OmniSciDB.

Reply:
Currently, there are two remote JIT classes:
- RemoteOmnisci as defined in omniscidb.py that produced LLVM IR should never contain Python C/API calls.
- RemoteJIT as defined in remotejit.py that LLVM IR *may* contain Python C/API calls.

TargetInfo has the following properties `has_cpython`, `has_numpy`, `has_numba` that should be used for determining which registries *can be* installed for target/typing contexts. In the future, omniscidb might be linking against Python libraries and the using Python C/API calls become allowed for omniscidb target/typing contexts as well.

So, I think in all cases we should use JITRemoteTargetContext/JITRemoteTypingContext but when installing additional libraries, take into account TargetInfo properties. I'll create a separate issue for addressing this problem.

Reply:
Removed the if-block as unnecessary: compile_to_LLVM should always use remote target/typing context that should never compile machine-level code for the host target.